ANSWERS.md

```diff
@@ -0,0 +1,21 @@
+## Self-Study/Essay Questions
+
+- [ ] What are PropTypes used for? Please describe why it's important to type check our data in JavaScript.
+
+    Proptypes are used to verify that certain values for variable matc the data type that you want for instance fir username you would expect a string.
+
+
+- [ ] Describe a life-cycle event in React?
+
+    Render is a lifecycle method that rerenders te page upon any changes to state. It is called as many times as there is a change to state.
+
+- [ ] Explain the details of a Higher Order Component?
+
+    A higher order component is component that takes in another component as its argument parameter
+
+
+- [ ] What are three different ways to style components in React? Explain some of the benefits of each.
+
+    Reactstrap: Very fast and fluid to incorporate/ useful for prototyping/ limited in use les creative freedom
+    CSS in JS components: Highly customizable and dry to apply styling on different objects in react
+    Less/Sass/compiler: Familiar basic, use of classnames to target styling
```

Comment:
4/4 , I think one of your answers for the last question could include styled-components